core/lease/store.go

```diff
@@ -19,17 +19,17 @@ type Store interface {
 	// it succeeds, the claim is guaranteed until at least the supplied duration
 	// after the call to ClaimLease was initiated. If it returns ErrInvalid,
 	// check Leases() for updated state.
-	ClaimLease(lease Key, request Request, stop <-chan struct{}) error
+	ClaimLease(lease Key, request Request) error
```

Comment:
I'm not 100% convinced by this. I love the simplification. But it does mean that you're stopping the request because the store is stopping, not because the caller is stopping. Which is at least semantically different.
If you feel it makes more sense this way, I'm ok with the change (because it is definitely a nice simplification).
I just wanted to point out that it *is* a change in semantic.